packages/react/src/App.tsx

```diff
@@ -1,42 +1,62 @@
 import "./assets/main.css";
-import { BrowserRouter, Route, Routes } from "react-router";
-import { ProjectTable } from "./components/ProjectTable";
-import { EntityView } from "./components/EntityView";
-import { FallbackPage } from "./components/FallbackPage";
-import { ProjectProvider } from "./contexts/Project/ProjectProvider";
-
-function App() {
-  console.log(import.meta.env);
+import { KEY_BINDINGS } from "./constants/keybindings";
+import { useKeybinding } from "./hooks/keyboardEventHandler";
+import React, { memo } from "react";
+import { createCommandCenter } from "./components/CommandCenter";
+import { AppRoutes, type AppRoutesProps } from "./routes";
+import { defaultSuggestions, InputAction, Suggestion } from "./constants/commandTypes";
+import { ProjectContextType } from "./contexts/Project/useProjectContext";
 
+const AppComponent = React.memo(function AppComponent({
+  AppRoutes,
+  CommandCenter
+}: {
+  AppRoutes: React.ComponentType<AppRoutesProps>,
+  CommandCenter: React.ComponentType
+}) {
   return (
-    <BrowserRouter>
-      <Routes>
-        <Route path="/:workspaceId/projects/:projectId">
-          {/* Table view (index route) */}
-          <Route
-            index
-            element={
-              <ProjectProvider>
-                <ProjectTable />
-              </ProjectProvider>
-            }
-          />
-
-          {/* Entity view */}
-          <Route
-            path="entities/:entityId"
-            element={
-              <ProjectProvider>
-                <EntityView />
-              </ProjectProvider>
-            }
-          />
-        </Route>
-        {/* Fallback route */}
-        <Route path="*" element={<FallbackPage />} />
-      </Routes>
-    </BrowserRouter>
+    <>
+      <AppRoutes CommandCenter={CommandCenter} />
+    </>
   );
-}
+});
+
+export const createApp = () => {
+
+  const fetchSuggestions = async (input: string, projectContext?: ProjectContextType): Promise<Suggestion[]> => {
+    if (!projectContext) return [];
+
+    /**
+     * We should add keywords and descriptions to each action
+     * Then we can use NLP here to find best matches
+     */
+
+    if (!input.trim()) {
+      return defaultSuggestions.filter(suggestion =>
+        suggestion.shouldShow(projectContext)
+      );
+    }
+
+    return defaultSuggestions.filter(suggestion =>
+      suggestion.shouldShow(projectContext) &&
+      suggestion.label.toLowerCase().includes(input.toLowerCase())
+    );
+  };
```

Comment:
The fetch suggestions has been abstracted away and can be made more powerful with NLP, fuzzy search or whatever we want to do.